packages/core/src/App/Components/Layout/Header/menu-link.tsx

```diff
@@ -0,0 +1,174 @@
+                    {suffix_icon && <Icon className='header__menu-mobile-link-suffix-icon' icon={suffix_icon} />}
+                </div>
+            );
+        }
+
+        if (!link_to) {
+            return (
+                <div
+                    className={classNames('header__menu-mobile-link', {
+                        'header__menu-mobile-link--disabled': is_disabled,
+                    })}
+                    data-testid={data_testid}
+                >
+                    <Icon className='header__menu-mobile-link-icon' icon={icon} />
+                    <span className='header__menu-mobile-link-text'>{text}</span>
+                    {suffix_icon && <Icon className='header__menu-mobile-link-suffix-icon' icon={suffix_icon} />}
+                </div>
+            );
+        } else if (is_external_link) {
+            return (
+                <a
+                    className={classNames('header__menu-mobile-link', {
+                        'header__menu-mobile-link--disabled': is_disabled,
+                        'header__menu-mobile-link--active': is_active,
+                    })}
+                    href={link_to}
+                    data-testid={data_testid}
+                >
+                    <Icon className='header__menu-mobile-link-icon' icon={icon} />
+                    <Text
+                        className={is_trade_text ? '' : 'header__menu-mobile-link-text'}
+                        as='h3'
+                        size='xs'
+                        weight={window.location.pathname === '/' && is_trade_text ? 'bold' : undefined}
+                    >
+                        {text}
+                    </Text>
+                    {suffix_icon && <Icon className='header__menu-mobile-link-suffix-icon' icon={suffix_icon} />}
+                </a>
+            );
+        }
+
+        return (
+            <BinaryLink
+                to={link_to}
+                className={classNames('header__menu-mobile-link', {
+                    'header__menu-mobile-link--disabled': is_disabled,
+                    'header__menu-mobile-link--active': is_active,
+                })}
+                active_class='header__menu-mobile-link--active'
+                onClick={onClickLink}
+                data-testid={data_testid}
+            >
+                <Icon className='header__menu-mobile-link-icon' icon={icon} />
+                <Text
+                    className={is_trade_text ? '' : 'header__menu-mobile-link-text'}
+                    as='h3'
+                    size='xs'
+                    weight={window.location.pathname === '/' && is_trade_text ? 'bold' : undefined}
+                >
+                    {text}
+                </Text>
+                {suffix_icon && <Icon className='header__menu-mobile-link-suffix-icon' icon={suffix_icon} />}
+            </BinaryLink>
+        );
+    }
+);
+
+export default MenuLink;
```

Comment:
@yauheni-deriv This component is doing lots of things, Can we break it down? 🤔

Reply:
@farzin-deriv, we can do it as improvements later, this card has been tested twice already :)
Let’s bring this functionality to production 
(btw, there is also important fix for reloading pages while navigating account menu in mobile :) )